plugins/search/src/components/SearchResult/SearchResult.tsx

```diff
@@ -209,13 +209,30 @@ export const SearchResult = ({ searchQuery }: SearchResultProps) => {
     }));
   };
 
+  const filterOptions = results.reduce(
+    (acc, curr) => {
+      if (curr.kind && acc.kind.indexOf(curr.kind) < 0) {
+        acc.kind.push(curr.kind);
+      }
+      if (curr.lifecycle && acc.lifecycle.indexOf(curr.lifecycle) < 0) {
+        acc.lifecycle.push(curr.lifecycle);
+      }
+      return acc;
+    },
+    {
+      kind: [] as Array<string>,
+      lifecycle: [] as Array<string>,
+    },
+  );
+
   return (
     <>
       <Grid container>
         {showFilters && (
           <Grid item xs={3}>
             <Filters
               filters={filters}
```

Comment:
Now when we have filterOptions as well, I think we can rename the `filters` state to `selectedFilters` or something more clear